word-search.py

```diff
@@ -212,9 +216,111 @@ def randomizer(grid):
 
     return grid
 
-def generateWordList():
-    # compiles the list of words that the user can find in the game
-    return
+# creates a dictionary of all distinct letters in the grid as keys
+# and the list of all their coordinate locations as the value
+def find_starting_coordinates(grid):
+    startingCoordinates = {}
+    for x in range(0, len(grid)):
+        for y in range(0, len(grid[x])):
+            if grid[x][y] in startingCoordinates.keys():
+                startingCoordinates[grid[x][y]].append([x, y])
+            else:
+                startingCoordinates[grid[x][y]] = [[x, y]]
+    return startingCoordinates
+
+def find_next_tile(nextLetter, xyCoordinates, grid, usedTiles):
+    x = xyCoordinates[0]
+    y = xyCoordinates[1]
+    # requested to make the code easier to read
+    up = x-1
+    down = x+1
+    left = y-1
+    right = y+1
+    maxIndex = len(grid)-1
+    matches = []
+
+    # checks each of the surround tiles if it exists,
+    # if it has not yet been used to form the word,
+    # and if is equal to the next letter in the key/word
+    if (up >= 0 and left >= 0 and (not [up, left] in usedTiles) and nextLetter == grid[up][left]):
```

Comment:
NOT A BUG! NOTE LANG

Kung okay lang 😬 I want to try exploring an alternative implementation for this as practice. Encountered from an online course lang dati yung idea na when something seems repeating parang pwede iloop. Will try exploring this as a challenge lang pag may free time :)

Reply:
Sure! Create another branch nalang din to implement this :)